### PR TITLE
:sparkles: Add `trigger_scheduler`

### PR DESCRIPTION
**Pull request by elbeno**

### CODE REVIEW

File: test/schedulers/trigger_scheduler.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          "[trigger_scheduler]") {
+    using S = async::trigger_scheduler<"name">;
+    auto s = S::schedule();
+    auto cs =
+        async::get_completion_scheduler<async::set_value_t>(async::get_env(s));
+    static_assert(std::same_as<decltype(cs), S>);
+}
+
+TEST_CASE("trigger_scheduler schedules tasks", "[trigger_scheduler]") {
+    auto s = async::trigger_scheduler<"name">{};
```
Comment (mjcaisse-intel) [inline test/schedulers/trigger_scheduler.cpp:37]: issue: using one static scheduler for all tests.. potential state entanglement
I think `"name"` (throughout file) needs to be some unique type trick (decltype-of-lambda) to ensure that tests do not interfere with each other.

**Review by mjcaisse-intel — COMMENTED** [review]
One update for the tests. This PR looks great! I'm looking forward to replacing my `interrupt_scheduler` with it as well as using it as the building block for communication channels.

File: include/async/schedulers/trigger_manager.hpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+using trigger_task = single_linked_task<task_base>;
+
+template <stdx::ct_string Name, triggerable_task Task = trigger_task>
+struct trigger_manager {
+    using task_t = Task;
+
+  private:
+    struct mutex;
+    std::array<stdx::intrusive_forward_list<task_t>, 1> tasks{};
```
[the remark this answers is not in the document]
Comment (elbeno) [reply include/async/schedulers/trigger_manager.hpp:34]: Yes :) Do you have a better idea?

[the remark this answers is not in the document]
Comment (mjcaisse-intel) [reply include/async/schedulers/trigger_manager.hpp:34]: Nope  (o;   Just making sure I understood.

File: test/schedulers/trigger_scheduler.cpp (hunk 1 above)
Comment (elbeno) [reply test/schedulers/trigger_scheduler.cpp]: Good call - I've disentangled them by using `TEMPLATE_TEST_CASE`s

**Review by mjcaisse-intel — APPROVED** [review]

**Pull request merged**